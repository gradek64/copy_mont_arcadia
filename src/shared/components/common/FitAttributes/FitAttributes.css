/** @define FitAttributes */

.FitAttributes {
  margin: $fit-attributes-margin;

  &-link {
    border: $fit-attributes-link-border;
    border-radius: $fit-attributes-link-border-radius;
    margin: $fit-attributes-link-margin;
    line-height: $fit-attributes-link-line-height;
    text-transform: $fit-attributes-link-text-transform;
    padding: $fit-attributes-link-text-padding;
    color: $fit-attributes-link-text-color;
    font-size: $fit-attributes-link-text-font-size;
    text-decoration: none;
    display: inline-block;
  }

  &--stateActive {
    color: $fit-attributes-link-font-color-state-active;
    background-color: $fit-attributes-link-background-color-state-active;
    border-color: $fit-attributes-link-border-color-state-active;
  }
}

@media $min-tablet {
  .FitAttributes {
    margin: $fit-attributes-margin-desktop;
  }
}
