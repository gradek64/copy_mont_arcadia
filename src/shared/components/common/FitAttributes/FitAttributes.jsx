import PropTypes from 'prop-types'
import React from 'react'
import { Link } from 'react-router'
import classNames from 'classnames'

import FeatureCheck from '../FeatureCheck/FeatureCheck'

const FitAttributes = ({
  fitAttributes,
  isQuickview,
  onClick,
  swatchChange,
}) => (
  <FeatureCheck flag="FEATURE_SHOW_FIT_ATTRIBUTE_LINKS">
    <div className="FitAttributes">
      {fitAttributes.map(
        (fitAttribute) =>
          isQuickview ? (
            <button
              onClick={() => {
                swatchChange()
                if (!fitAttribute.isTPMActive) {
                  onClick(fitAttribute.catentryId)
                }
              }}
              key={fitAttribute.catentryId}
              className={classNames('FitAttributes-link', {
                'FitAttributes--stateActive': fitAttribute.isTPMActive,
              })}
            >
              {fitAttribute.TPMName}
            </button>
          ) : (
            <Link
              className={classNames('FitAttributes-link', {
                'FitAttributes--stateActive': fitAttribute.isTPMActive,
              })}
              onClick={() => swatchChange()}
              to={fitAttribute.TPMUrl}
              key={fitAttribute.catentryId}
            >
              {fitAttribute.TPMName}
            </Link>
          )
      )}
    </div>
  </FeatureCheck>
)

FitAttributes.propTypes = {
  fitAttributes: PropTypes.array.isRequired,
  isQuickview: PropTypes.bool,
  onClick: PropTypes.func,
  swatchChange: PropTypes.func,
}

FitAttributes.defaultProps = {
  isQuickview: false,
  onClick: () => {},
  swatchChange: () => {},
}

export default FitAttributes
