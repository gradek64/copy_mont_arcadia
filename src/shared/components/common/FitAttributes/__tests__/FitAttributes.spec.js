import renderComponentHelper from '../../../../../../test/unit/helpers/test-component'
import FitAttributes from '../FitAttributes'
import { Link } from 'react-router'

const renderComponent = renderComponentHelper(FitAttributes)

const initialProps = {
  fitAttributes: [
    {
      catentryId: '20434610',
      TPMName: 'Tall',
      isTPMActive: true,
      TPMUrl: '/en/tsuk/product/moto-star-stud-denim-borg-jacket-7196268',
    },
    {
      catentryId: '20906473',
      TPMName: 'Petite',
      isTPMActive: false,
      TPMUrl: '/en/tsuk/product/moto-knot-tie-jumpsuit-7178618',
    },
    {
      catentryId: '20865068',
      TPMName: 'Maternity',
      isTPMActive: false,
      TPMUrl: '/en/tsuk/product/moto-side-striped-denim-skirt-7216592',
    },
  ],
}

describe('<FitAttributes/>', () => {
  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent(initialProps).getTree()).toMatchSnapshot()
    })

    describe('not quickview', () => {
      it('renders the correct amount of fit attribute links', () => {
        const { wrapper } = renderComponent(initialProps)
        expect(wrapper.find('.FitAttributes-link').length).toBe(3)
      })

      it('first `Link` item has class of `FitAttributes--stateActive`', () => {
        const { wrapper } = renderComponent(initialProps)
        expect(
          wrapper
            .find(Link)
            .first()
            .hasClass('FitAttributes--stateActive')
        ).toBe(true)
      })

      it('only one `Link` item has class of `FitAttributes--stateActive`', () => {
        const { wrapper } = renderComponent(initialProps)
        expect(wrapper.find('.FitAttributes--stateActive').length).toBe(1)
      })
    })
    describe('quickview', () => {
      it('renders the correct amount of fit attribute buttons', () => {
        const { wrapper } = renderComponent({
          ...initialProps,
          isQuickview: true,
        })
        expect(wrapper.find('.FitAttributes-link').length).toBe(3)
      })

      it('first item has class of `FitAttributes--stateActive`', () => {
        const { wrapper } = renderComponent({
          ...initialProps,
          isQuickview: true,
        })
        expect(
          wrapper
            .find('.FitAttributes-link')
            .first()
            .hasClass('FitAttributes--stateActive')
        ).toBe(true)
      })

      it('only one item has class of `FitAttributes--stateActive`', () => {
        const { wrapper } = renderComponent({
          ...initialProps,
          isQuickview: true,
        })
        expect(wrapper.find('.FitAttributes--stateActive').length).toBe(1)
      })
    })
  })
  describe('@events', () => {
    describe('isQuickview is true', () => {
      it('should not call onClick when isTPMActive is true', () => {
        const props = {
          ...initialProps,
          isQuickview: false,
          onClick: jest.fn(),
        }
        const { wrapper } = renderComponent(props)

        wrapper
          .find('.FitAttributes-link')
          .first()
          .simulate('click')
        expect(props.onClick).not.toHaveBeenCalled()
      })
      it('should call onClick when isTPMActive is false', () => {
        const props = {
          ...initialProps,
          isQuickview: true,
          onClick: jest.fn(),
        }
        const { wrapper } = renderComponent(props)

        wrapper
          .find('.FitAttributes-link')
          .at(1)
          .simulate('click')
        expect(props.onClick).toHaveBeenCalled()
      })
      it('should call swatchChange', () => {
        const props = {
          ...initialProps,
          isQuickview: false,
          swatchChange: jest.fn(),
        }
        const { wrapper } = renderComponent(props)

        // works irrespective of isTMPActive prop
        wrapper
          .find('.FitAttributes-link')
          .at(0)
          .simulate('click')
        wrapper
          .find('.FitAttributes-link')
          .at(1)
          .simulate('click')
        expect(props.swatchChange).toHaveBeenCalledTimes(2)
      })
    })
    describe('isQuickview is false', () => {
      it('should not call onClick regardless of isTMPActive attribute', () => {
        const props = {
          ...initialProps,
          isQuickview: false,
          onClick: jest.fn(),
        }

        const { wrapper } = renderComponent(props)

        wrapper
          .find('.FitAttributes-link')
          .at(0)
          .simulate('click')

        wrapper
          .find('.FitAttributes-link')
          .at(1)
          .simulate('click')

        expect(props.onClick).toHaveBeenCalledTimes(0)
      })
      it('should call swatchChange', () => {
        const props = {
          ...initialProps,
          isQuickview: false,
          swatchChange: jest.fn(),
        }

        const { wrapper } = renderComponent(props)

        wrapper
          .find('.FitAttributes-link')
          .at(0)
          .simulate('click')

        wrapper
          .find('.FitAttributes-link')
          .at(1)
          .simulate('click')

        expect(props.swatchChange).toHaveBeenCalledTimes(2)
      })
    })
  })
})
