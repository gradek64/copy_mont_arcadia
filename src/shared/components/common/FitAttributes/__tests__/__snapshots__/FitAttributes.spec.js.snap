// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FitAttributes/> @renders in default state 1`] = `
<Connect(FeatureCheck)
  flag="FEATURE_SHOW_FIT_ATTRIBUTE_LINKS"
>
  <div
    className="FitAttributes"
  >
    <Link
      className="FitAttributes-link FitAttributes--stateActive"
      key="20434610"
      onClick={[Function]}
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/en/tsuk/product/moto-star-stud-denim-borg-jacket-7196268"
    >
      Tall
    </Link>
    <Link
      className="FitAttributes-link"
      key="20906473"
      onClick={[Function]}
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/en/tsuk/product/moto-knot-tie-jumpsuit-7178618"
    >
      Petite
    </Link>
    <Link
      className="FitAttributes-link"
      key="20865068"
      onClick={[Function]}
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/en/tsuk/product/moto-side-striped-denim-skirt-7216592"
    >
      Maternity
    </Link>
  </div>
</Connect(FeatureCheck)>
`;
